platform/view/services/kvs/kvs.go

```diff
@@ -123,33 +123,36 @@ func (o *KVS) Exists(id string) bool {
 }
 
 func (o *KVS) Put(id string, state interface{}) error {
-	o.putMutex.Lock()
-	defer o.putMutex.Unlock()
-
 	raw, err := json.Marshal(state)
 	if err != nil {
 		return errors.Wrapf(err, "cannot marshal state with id [%s]", id)
 	}
 
-	if err := o.store.BeginUpdate(); err != nil {
+	tx, err := o.store.NewWriteTransaction()
+	if err != nil {
 		return errors.Wrapf(err, "begin update for id [%s] failed", id)
 	}
 
 	logger.Debugf("store [%d] bytes into key [%s:%s]", len(raw), o.namespace, id)
-	if err := o.store.SetState(o.namespace, id, raw); err != nil {
-		if err1 := o.store.Discard(); err1 != nil {
+	if err := tx.SetState(o.namespace, id, raw); err != nil {
+		if err1 := tx.Discard(); err1 != nil {
 			logger.Debugf("got error %v; discarding caused %v", err, err1)
 		}
 
 		if !errors.HasCause(err, driver.UniqueKeyViolation) {
 			return errors.Wrapf(err, "failed to commit value for id [%s]", id)
 		}
 	} else {
-		if err := o.store.Commit(); err != nil {
+		if err := tx.Commit(); err != nil {
```

Comment:
We're only doing a single SetState in this transaction. Why do we need a TransactionalUnversionedPersistence? I noticed we do it in Delete as well, and also in NewSimpleTXIDStore and in the vault's SetStatus. Wouldn't it be better to just fire the SetState directly in those cases?

Reply:
it is to allow parallel puts

Reply:
As discussed offline, I don't think it's necessary for the sql drivers, but we'll handle it in a new issue / PR.